### ISSUE TITLE
Query router for multiple cluster setup?

### ISSUE THREAD

**Issue by tesseract2048** [OpenTSDB/opentsdb#929]
Monitoring is something worth multiple-datacenter deployment. In which case, theoretically we can tolerate (N-1) DC failure.
This probably requires a proxy-alike mechanism to serve as front-end for queries while managing datacenter routing and load balance. However, I doubt the mechanism can be very different among users. Is there anything general we can do about that?

**Comment by manolama** [comment]
Yeah, we have a query layer similar to Splicer at Yahoo that is making its way upstream slowly (finishing up the query pipeline now). We dual-write and dual-query all of our data and the query layer does the balancing and merging. Hoping to have it out by summer of this year.
